tron/config/schema.py

```diff
@@ -26,9 +26,12 @@ def config_object_factory(name, required=None, optional=None):
     """
     required = required or []
     optional = optional or []
+
     config_class = namedtuple(name, required + optional)
+    config_class.__new__.__defaults__ = (None,) * len(optional)
```

Comment:
Hmm. Is this to "DRY" out a lot of the defaults code stuff? The line itself looks magic to me. Can you add a... comment?

Reply:
sure, i stole it from SO. it just makes last `len(optional)` arguments... optional :) kinda surprised namedtuple doesn't have a more convenient way to achieve this.